wrapper/src/main/java/software/amazon/jdbc/plugin/cache/CacheConnection.java

```diff
@@ -277,6 +290,45 @@ public CacheConnection(final Properties properties) {
     this(properties, null);
   }
 
+  /**
+   * Detects whether the Redis endpoint is running in cluster mode by executing INFO command.
+   * Caches the result to avoid repeated detection.
+   */
+  private void detectClusterMode() {
+    if (isClusterMode != null) {
+      return;
+    }
+
+    synchronized (this) {
+      if (isClusterMode != null) {
+        return;
+      }
+
+      String[] hostnameAndPort = getHostnameAndPort(cacheRwServerAddr);
+      RedisURI redisUri = buildRedisURI(hostnameAndPort[0], Integer.parseInt(hostnameAndPort[1]));
+
+      ClientResources resources = ClientResources.builder().build();
+
+      try (RedisClient client = RedisClient.create(resources, redisUri);
+           StatefulRedisConnection<byte[], byte[]> conn = client.connect(new ByteArrayCodec())) {
+
+        String infoOutput = conn.sync().info("cluster");
+        boolean clusterEnabled = infoOutput != null && infoOutput.contains("cluster_enabled:1");
```

Comment:
Better to look for `cluster_enabled` field and check its value instead of looking for `cluster_enabled:1`.